riot/riot.py

```diff
@@ -246,31 +246,28 @@ class VenvInstance:
     parent: t.Optional["VenvInstance"] = None
 
     @property
-    def venv_path(self) -> t.Optional[str]:
-        """Return path to directory of the venv this instance should use.
+    def prefix(self) -> t.Optional[str]:
+        """Return path to directory where dependencies should be installed.
 
-        This will return a python version + package specific venv path name.
-        If no packages are defined it will return the ``Interpreter.venv_path``.
+        This will return a python version + package specific path name.
+        If no packages are defined it will return ``None``.
         """
-        if self.py is None:
+        if self.py is None or not self.pkgs:
             return None
 
         venv_path = self.py.venv_path
-        if self.needs_venv:
-            venv_path = "_".join(
-                [venv_path] + [f"{n}{rmchars('<=>.,', v)}" for n, v in self.pkgs]
-            )
-        return venv_path
+        assert venv_path is not None, self
 
-    @property
-    def target(self) -> t.Optional[str]:
-        """Target path for package installation."""
-        return self.venv_path
+        ident = self.ident
+        assert ident is not None, self
+        return "_".join((venv_path, ident))
 
     @property
-    def needs_venv(self) -> bool:
-        """Whether this ``VenvInstance`` needs its own venv or not."""
-        return bool(self.pkg_str)
+    def ident(self) -> t.Optional[str]:
+        """Return prefix identifier string based on packages."""
+        if not self.pkgs:
+            return None
+        return "_".join((f"{n}{rmchars('<=>.,', v)}" for n, v in self.pkgs))
```

Comment:
crazy idea... I wonder if we should normalize like `>=` to `gte_` so `flask>=1.0.0` -> `flask_gte_100`

also, are we going to have issues with `flask>=1.1.0` and `flask>=1.10.0` ?

(this is an aside, non-blocking for this PR

Reply:
That's a good point. The more unique we make this the better! 🙂

Reply:
also `~`!

Reply:
`flask~=1.0.0` -> `flask_almost_100` 😆